test/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+#
+# Distributed Linear Algebra with Future (DLAF)
+#
+# Copyright (c) 2018-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+set(DLAF_FORTRAN_MPIRUN "" CACHE STRING "MPI run command.")
```

Comment:
is there any reason why it is better to have this custom variable instead of using `FindMPI` [variables](https://cmake.org/cmake/help/latest/module/FindMPI.html#variables-for-locating-mpi)?

Reply:
It's mostly for convenience, since the `FindMPI` variables are a mouthful.

```
-DDLAF_FORTRAN_MPIRUN="mpiexec -n 6"
```

vs

```
-DMPIEXEC_EXECUTABLE="mpiexec" -DMPIEXEC_NUMPROC_FLAG="-n" -DMPIEXEC_MAX_NUMPROCS="6"
```

I can change it if it's preferred, but if some of those variables are set automatically, it might interfere with CI.

Reply:
I changed the CMake logic to allow the use of those variables directly. Ultimately, only `MPIEXEC_MAX_NUMPROCS` needs to be set explicitly. I included a check to ensure that it is set to `6`, which is the number of ranks that the tests require.

I also added a `DLAF_FORTRAN_CSCS_CI` variable (conditional on testing being enabled) and a `+cscs-ci` variant in the Spack package (only for the local recipe) that does the right thing for CI.